<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ActionWizard</class>
 <widget class="QWizard" name="ActionWizard">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>629</width>
    <height>670</height>
   </rect>
  </property>
  <property name="windowIcon">
   <iconset theme="alterator"/>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <property name="options">
   <set>QWizard::WizardOption::NoBackButtonOnStartPage</set>
  </property>
  <widget class="QWizardPage" name="initialPage">
   <attribute name="pageId">
    <string notr="true">0</string>
   </attribute>
   <layout class="QVBoxLayout" name="verticalLayout_4" stretch="0,1,0,0,0,0,0">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Policy::Expanding</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QGroupBox" name="resourcesGroupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Resources</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_12">
       <item>
        <widget class="QLabel" name="deployResourceMessage">
         <property name="text">
          <string>The following system resources will be owned by this service after deployment:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="undeployResourceMessage">
         <property name="text">
          <string>The following system resources will be released after undeploying this service:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="CustomTreeView" name="initialResourceOwnersView"/>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="KMessageWidget" name="conflictsDetectedMessage">
      <property name="text">
       <string>Some resources are owned by another deployed services.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="closeButtonVisible">
       <bool>false</bool>
      </property>
      <property name="messageType">
       <enum>KMessageWidget::MessageType::Warning</enum>
      </property>
      <property name="icon">
       <iconset theme="dialog-warning"/>
      </property>
     </widget>
    </item>
    <item>
     <widget class="KMessageWidget" name="conflictsResolvableMessage">
      <property name="text">
       <string>All resource conflicts may be resolved by overriding these resources on the &quot;Parameters&quot; page.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="closeButtonVisible">
       <bool>false</bool>
      </property>
      <property name="messageType">
       <enum>KMessageWidget::MessageType::Information</enum>
      </property>
      <property name="icon">
       <iconset theme="dialog-ok"/>
      </property>
     </widget>
    </item>
    <item>
     <widget class="KMessageWidget" name="conflictsUnresolvableMessage">
      <property name="text">
       <string>Deployment is not possible, because these resources could not be overridden.
You may want to undeploy the conflicting services, or reconfigure them, if their resourses are overridable.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="closeButtonVisible">
       <bool>false</bool>
      </property>
      <property name="messageType">
       <enum>KMessageWidget::MessageType::Error</enum>
      </property>
      <property name="icon">
       <iconset theme="dialog-error"/>
      </property>
     </widget>
    </item>
    <item>
     <widget class="KMessageWidget" name="conflictsOkMessage">
      <property name="text">
       <string>Conflicts were not detected.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="closeButtonVisible">
       <bool>false</bool>
      </property>
      <property name="messageType">
       <enum>KMessageWidget::MessageType::Positive</enum>
      </property>
      <property name="icon">
       <iconset theme="dialog-ok"/>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="optionsGroupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Additional options</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <widget class="QCheckBox" name="forceDeployCheckBox">
         <property name="toolTip">
          <string>When this option is activated, diagnostic tests are not available. The service will be deployed regardless of the current status.</string>
         </property>
         <property name="text">
          <string>Force deploy (skip premilinary tests and ignore current status)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="otherOptionsContainer" native="true">
         <layout class="QVBoxLayout" name="verticalLayout_13">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QCheckBox" name="autoStartCheckBox">
            <property name="toolTip">
             <string>Service will be started immediately after a successfull deploy</string>
            </property>
            <property name="text">
             <string>Start service immediately after deploy</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="finishPageCheckBox">
            <property name="toolTip">
             <string>An additional page will be shown where you can double-check your selection.</string>
            </property>
            <property name="text">
             <string>Check the selected parameters before applying</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="preDiagPage">
   <property name="title">
    <string>Premilinary diagnostics</string>
   </property>
   <property name="subTitle">
    <string>Run diagnostic tests to determine if deployment is currently possible</string>
   </property>
   <attribute name="pageId">
    <string notr="true">1</string>
   </attribute>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QCheckBox" name="preDiagCheckBox">
      <property name="text">
       <string>Enable</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="DiagSelectionForm" name="preDiagSelector" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="postDiagPage">
   <property name="title">
    <string>Post-deploy diagnostics</string>
   </property>
   <property name="subTitle">
    <string>Run diagnostic tests to check deployed service</string>
   </property>
   <attribute name="pageId">
    <string notr="true">2</string>
   </attribute>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QCheckBox" name="postDiagCheckBox">
      <property name="text">
       <string>Enable</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="DiagSelectionForm" name="postDiagSelector" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="parametersPage">
   <property name="title">
    <string>Deployment parameters</string>
   </property>
   <property name="subTitle">
    <string>Set deployment configuration</string>
   </property>
   <attribute name="pageId">
    <string notr="true">3</string>
   </attribute>
   <layout class="QVBoxLayout" name="verticalLayout_7" stretch="0,1,0">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="SearchBar" name="searchBar" native="true"/>
    </item>
    <item>
     <layout class="QVBoxLayout" name="formContainer"/>
    </item>
    <item>
     <widget class="KMessageWidget" name="invalidParameterWarning">
      <property name="text">
       <string notr="true">Invalid parameter message</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="closeButtonVisible">
       <bool>false</bool>
      </property>
      <property name="messageType">
       <enum>KMessageWidget::MessageType::Error</enum>
      </property>
      <property name="icon">
       <iconset theme="dialog-error"/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="finishPage">
   <property name="title">
    <string>Deploy service</string>
   </property>
   <property name="subTitle">
    <string>You may want to double-check your selection</string>
   </property>
   <attribute name="pageId">
    <string notr="true">4</string>
   </attribute>
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="preDiagTab">
       <attribute name="icon">
        <iconset theme="system-run"/>
       </attribute>
       <attribute name="title">
        <string>Premilinary diagnostics</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <widget class="CustomTreeView" name="preDiagView">
          <property name="editTriggers">
           <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
          </property>
          <property name="showDropIndicator" stdset="0">
           <bool>false</bool>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SelectionMode::NoSelection</enum>
          </property>
          <attribute name="headerVisible">
           <bool>false</bool>
          </attribute>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="postDiagTab">
       <attribute name="icon">
        <iconset theme="system-run"/>
       </attribute>
       <attribute name="title">
        <string>Post-deploy diagnostics</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_10">
        <item>
         <widget class="CustomTreeView" name="postDiagView">
          <property name="editTriggers">
           <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
          </property>
          <property name="showDropIndicator" stdset="0">
           <bool>false</bool>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SelectionMode::NoSelection</enum>
          </property>
          <attribute name="headerVisible">
           <bool>false</bool>
          </attribute>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="parametersTab">
       <attribute name="icon">
        <iconset theme="preferences-system"/>
       </attribute>
       <attribute name="title">
        <string>Parameters</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_11">
        <item>
         <widget class="CustomTreeView" name="parametersView"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="resourcesTab">
       <attribute name="icon">
        <iconset theme="changes-prevent"/>
       </attribute>
       <attribute name="title">
        <string>Resources</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="CustomTreeView" name="confirmResourceOwnersView"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="KMessageWidget" name="missingDiagWarning">
      <property name="text">
       <string>Some of the diagnostic tools are missing! Test results may not be reliable.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="closeButtonVisible">
       <bool>false</bool>
      </property>
      <property name="messageType">
       <enum>KMessageWidget::MessageType::Warning</enum>
      </property>
      <property name="icon">
       <iconset theme="dialog-warning"/>
      </property>
     </widget>
    </item>
    <item>
     <widget class="KMessageWidget" name="missingDiagParamsWarning">
      <property name="text">
       <string>Some of the diagnostic tools take into account only a part of the parameters, that are nescessary for diagnostics. Test results may not be reliable.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="closeButtonVisible">
       <bool>false</bool>
      </property>
      <property name="messageType">
       <enum>KMessageWidget::MessageType::Warning</enum>
      </property>
      <property name="icon">
       <iconset theme="dialog-warning"/>
      </property>
     </widget>
    </item>
    <item>
     <widget class="KMessageWidget" name="forceDeployMessage">
      <property name="text">
       <string>Force deploy enabled. Service will be deployed regardless of the current status.</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="closeButtonVisible">
       <bool>false</bool>
      </property>
      <property name="messageType">
       <enum>KMessageWidget::MessageType::Warning</enum>
      </property>
      <property name="icon">
       <iconset theme="dialog-warning"/>
      </property>
     </widget>
    </item>
    <item>
     <widget class="KMessageWidget" name="autoStartMessage">
      <property name="text">
       <string>Service will be started immediately after a successfull deploy</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="closeButtonVisible">
       <bool>false</bool>
      </property>
      <property name="messageType">
       <enum>KMessageWidget::MessageType::Information</enum>
      </property>
      <property name="icon">
       <iconset theme="dialog-information"/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="progressPage">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <widget class="LogWidget" name="logWidget" native="true"/>
    </item>
    <item>
     <widget class="QProgressBar" name="progressBar">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximum">
       <number>0</number>
      </property>
      <property name="value">
       <number>-1</number>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KMessageWidget</class>
   <extends>QFrame</extends>
   <header>kmessagewidget.h</header>
  </customwidget>
  <customwidget>
   <class>CustomTreeView</class>
   <extends>QTreeView</extends>
   <header>ui/CustomTreeView.h</header>
  </customwidget>
  <customwidget>
   <class>DiagSelectionForm</class>
   <extends>QWidget</extends>
   <header>ui/DiagSelectionForm.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>LogWidget</class>
   <extends>QWidget</extends>
   <header>ui/LogWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SearchBar</class>
   <extends>QWidget</extends>
   <header>ui/SearchBar.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>preDiagCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>preDiagSelector</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>81</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>77</x>
     <y>179</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>postDiagCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>postDiagSelector</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>82</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>83</x>
     <y>162</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>forceDeployCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>forceDeployMessage</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>291</x>
     <y>575</y>
    </hint>
    <hint type="destinationlabel">
     <x>291</x>
     <y>435</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>autoStartCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>autoStartMessage</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>291</x>
     <y>583</y>
    </hint>
    <hint type="destinationlabel">
     <x>291</x>
     <y>485</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
